zetasql-toolkit-bigquery/src/main/java/com/google/zetasql/toolkit/catalog/bigquery/BigQueryCatalog.java

```diff
@@ -576,7 +631,16 @@ public void addAllTablesUsedInQuery(String query, AnalyzerOptions options) {
             .map(tablePath -> String.join(".", tablePath))
             .filter(BigQueryReference::isQualified) // Remove non-qualified tables
             .collect(Collectors.toSet());
-    this.addTables(ImmutableList.copyOf((tables)));
+    Set<String> standardTables =
```

Comment:
We're creating 4 different copies of the `tables` set here, let's create only one